mozilla-services/syncstorage-rs
# feat: Add `--abort` and `--user_range` flags

## Description

* --abort stops copying BSO records after N instances.
* --user_range limits copy to offset:limit users.
* optionally sorts users by fxa_uid via `--sort_users` flag
* collapses previous pushes and syncs with master branch

## Testing

* configure local environment using the steps provided in the `tools/README.md` file
* select a contiguous range of users specified in the given node starting from a given offset (e.g. if a node contains 100 users, you wish to copy 20 users, and you wish to skip the first 10 users, use a `user_range=10:20`. Verify users being copied fit the specified range.
* for a given category, select a subset of records you wish to copy over. Other records are dropped. (e.g. if a user has 100 history items and you wish to test that only 10 are copied over, use `abort=history:10`). App reports rows added or skipped.
* Can use `--dryrun` which will not write data to spanner, but act as if it did. 

## Issue(s)

Issue #407
